.github/workflows/karate.yml

```diff
@@ -29,11 +34,20 @@ jobs:
       - name: Deploy
         run: sam deploy --no-confirm-changeset --no-fail-on-empty-changeset  --stack-name ${{env.STACK_NAME}} --parameter-overrides ParameterKey=CustomDomain,ParameterValue=${DOMAIN_NAME} ParameterKey=CustomDomainBasePath,ParameterValue=${{env.BASE_PATH}} ParameterKey=CristinApiUrl,ParameterValue=${CRISTIN_API_URL}
       # Take a break waiting for API to deploy
-      - name: Sleep for 30 seconds
-        run: sleep 30s
+      - name: Sleep for 10 seconds
+        run: sleep 10s
+      # Create tet users in cognito
+      - name: Create karate test users
+        run: BASE_PATH=${{env.BASE_PATH}} ./gradlew createTestUsers -x karateTest -x test --info
+      # Take a break waiting for users
+      - name: Sleep for 10 seconds
+        run: sleep 10s
       # Run karate tests on deployed api
       - name: Run karate tests with Gradle
         run: BASE_PATH=${{env.BASE_PATH}} ./gradlew karateTest -x test --info
+        # Delete tests users
+      - name: Delete karate test users
```

Comment:
If karate tests fail will deletion of test user be executed? Or will we have a potential elevated user in Cognito?

Reply:
User and stack remain when tests crash - as for now it's good for PMD...